### PR TITLE
General HTML Template

### PR DESCRIPTION
**Pull request by 0Arian**
## What is this?

This is the base graphite program. It currently takes in no arguments but arguments will be added in the future. Currently it runs off a HTML template and renders that into a nice png file in your home folder

## How do I run this?

You need pip3 (python3's pip) and npm installed
with npm install

```
npm install -g phantomjs
```

with pip install the following dependencies
```
pip3 install click
pip3 install google-api-python-client
pip3 install jinja2
```

Make sure you have these environment variables set
```
VIEW_ID
CLIENT_SECRETS_PATH
```
To create these credentials follow this guide:
```
**NOTE; When prompted click Furnish a new private key and for the Key type select JSON, and save the generated key as client_secrets.json; you will need it later**
1. Go to https://console.developers.google.com/permissions/serviceaccounts
2. Click create service account
3. In the Create service account window, type a name for the service account, and select Furnish a new private key. Then click Create.
4. Your new public/private key pair is generated and downloaded to your machine; it serves as the only copy of this key. You are responsible for storing it securely.
5. Make sure you have your client_secrets.json file and set its path in the CLIENT_SECRETS_PATH environment variable
6. For your VIEW_ID you can find it under https://ga-dev-tools.appspot.com/account-explorer/, so set the VIEW_ID environmental variable with the #.
```
Now to run this just simply run the generate_report.py file
```
python3 generate_report.py
```
And a analytics.png file should appear in your directory with the rendered template

## Screenshot of example output
ScreenShot

### CODE REVIEW

File: generate_report.py [hunk 1]
```diff
@@ -1,20 +1,268 @@
...
+import argparse
+import os
+import sys
+import subprocess
+
+from apiclient.discovery import build
+import httplib2
+from oauth2client import client
+from oauth2client import file
+from oauth2client import tools
```
Comment (davidawad) [inline generate_report.py]: change this to import client, file, tools

File: generate_report.py [hunk 2]
```diff
@@ -1,20 +1,268 @@
...
+
+from jinja2 import Environment, FileSystemLoader
 
 import click
```
Comment (davidawad) [inline generate_report.py]: another style thing, but move all the single package imports to the top of the file. 

So put import click on top.

File: generate_report.py [hunk 3]
```diff
@@ -1,20 +1,268 @@
...
+        number_of_sessions=sessionNum
+        , number_of_users=userNum
+        , number_of_pageViews=pageViews
+        , bounce_rate_percentage=bounceRate
+        , daily_sessions=sessionCount
+        , engagement_session_channels=sessionChannels
+        , engagement_pageview_channels=pageviewChannels
+        , device_types=deviceTypes
+        , session_duration=sessionDuration
+        , country_per_session=countryPerSession
```
Comment (davidawad) [inline generate_report.py]: style: move these commas to the previous lines

**Comment by davidawad** [comment]
overall it looks pretty solid! just a couple of stylistic complaints from me

**Comment by 0Arian** [comment]
Alright all stylistic complaints dealt with and signatures have now been added to the project
You can input in a signature with the `--name` command line option

**Pull request merged**